thunder/clang/__init__.py

```diff
@@ -2055,6 +2055,28 @@ def zeta(a, b):
     )
 
 
+#
+# Elementwise ternary operations
+#
+
+
+@clangop()
+def lerp(start, end, weight):
+    inputs = (start, end, weight)
+    # torch.lerp does not promote types and only accepts floating-point inputs
+    computation_dtype, result_dtype = utils.elementwise_type_promotion(
+        *inputs, type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
+    )
+
+    inputs = maybe_broadcast(*inputs)
```

Comment:
What happens when `maybe_broadcast()` is called and weight is a number?

Reply:
It returns inputs without the `.shape` attribute untouched. 
https://github.com/Lightning-AI/lightning-thunder/blob/339a782e3d75061a065a3d2e47b5206f23aea7c3/thunder/clang/__init__.py#L1388-L1404

Example:
```py
import torch
import thunder
from thunder.clang import maybe_broadcast

@thunder.jit
def f(x, y, z):
    return maybe_broadcast(x, y, z)

f(torch.tensor([0.0, 1.0]), torch.tensor([2.0]), 0.4)
print(thunder.last_traces(f)[-1])
```
```py
@torch.no_grad()
@no_autocast
def computation(x, y):
  # x: "cpu f32[2]"
  # y: "cpu f32[1]"
  t1 = Tensor.expand(y, (2,))  # t1: "cpu f32[2]"
    # t1 = ltorch.expand(y, (2,))  # t1: "cpu f32[2]"
      # t1 = prims.broadcast_in_dim(y, (2,), (0,))  # t1: "cpu f32[2]"
  del y
  return (x, t1, 0.4)
```